spec/integration/common.js

```diff
@@ -0,0 +1,74 @@
+import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
+import $, { selector } from "./elements";
+import { version } from '../../package.json';
+
+const PORT = Cypress.env('PORT');
+
+export const click = (element) => {
+  $(element).click();
+};
+
+export const select = (value, element) => {
+  $(element).select(value);
+};
+
+export const type = (text, element) => {
+  $(element).type(text, { force: true }).trigger('input');
+};
+
+export const popupConfirm = (type) => {
+  let selector = '#default-confirm-popup-body .yes';
```

Comment:
I really suggest avoiding using CSS selectors and tags for selecting element. It makes them quite brittle to changes (for example changing the CSS classes or HTML element). Another approach is to use data-* attributes with randomly generated IDs, with the sole responsibility for being used in the ATs. It's then also helpful to store these inside a single config file with more identifiable names like ...

```json
{ 
  "WorkflowEditorConfirmButton": "ad5c7bad1f57"
}
```

```jsx
import atids from '../path/to/atids.json';

<button data-at-id={ atids.WorkflowEditorConfirmButton }>Confirm</button>
```

I use  the below to generate IDs. 

```
cat /dev/random | head -1 | shasum | cut -c 1-12
```

Reply:
totally agree with you and CSS selectors are not ideal, but we wanted to balance between effort and result. Editor was not prepared for testing and most of elements can not be identified easily. However theory we wanted to follow is not to touch the editor only after we have e2e coverage. If we wanted to go with test data attributes we need to touch the editor now and not sure if we wanna risk that. So I'd agree with this but maybe in the next round. WDYT?

Reply:
I think it's a pattern that shouldn't really be started, if the intention is to not use it going forward that is. Adding data attributes to the elements shouldn't cause any issues, and if it did it should be picked up by the tests in this PR 🙃 

I'm just expressing my opinion though and wouldn't think of block anything, it's absolutely your call.

Reply:
Problem is I'm not sure how to use these identifiers easily in angular templates 🤔 React is ok but in angular tempaltes we don't really have an easy way to use data from a simple json

Reply:
Ah ok, sorry I don't know about that. It's been a few years since I've done any Angular.

Reply:
ok let me investigate this, there is might be an easy way but definitely more complex than with react, so it'd be a follow up task